tests/test_generic_frontend.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Optional, Tuple, Sequence, Union, Callable, List
+
+import torch
+from torch import nn
+
+from i6_models.parts.frontend.generic_frontend import FrontendLayerType, GenericFrontendV1, GenericFrontendV1Config
+
+
+@dataclass
+class GenericFrontendV1TestParams:
+    batch: int
+    time: int
+    in_features: int
+    layer_ordering: Sequence[FrontendLayerType]
+    conv_kernel_sizes: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    conv_strides: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    conv_paddings: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    conv_out_dims: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    pool_kernel_sizes: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    pool_strides: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    pool_paddings: Optional[Sequence[Union[int, Tuple[int, int]]]]
+    activations: Optional[Sequence[Union[nn.Module, Callable[[torch.Tensor], torch.Tensor]]]]
+    out_features: int
```

Comment:
```suggestion
    cfg: GenericFrontendV1Config,
```
And modify the rest accordingly.